src/lib/videotagging_extensions/detection.js

```diff
@@ -0,0 +1,193 @@
+const DetectionAlgorithmManager = require('../detection_algorithms').DetectionAlgorithmManager;
+
+function Detection(videotagging, visitedFrames) {
+    this.videotagging = videotagging;
+    this.visitedFrames = visitedFrames;
+    this.detectionAlgorithmManager = new DetectionAlgorithmManager();
+
+    var self = this;
+
+    //maps every frame in the video to an imageCanvas NOTE mapVideo clears the oncanplay eventListener
+    this.mapVideo = function(frameHandler, exportUntil) {
+        return new Promise((resolve, reject) => {
+            //init canvas buffer
+            var frameCanvas = document.createElement("canvas");
+            frameCanvas.width = self.videotagging.video.videoWidth;
+            frameCanvas.height = self.videotagging.video.videoHeight;
+            var canvasContext = frameCanvas.getContext("2d");
+
+            // start exporting frames using the canplay eventListener
+            self.videotagging.video.oncanplay = iterateFrames;
+            self.videotagging.video.currentTime = 0;
+            self.videotagging.playingCallback();
+
+            function iterateFrames() {
+                var frameId = self.videotagging.getCurrentFrame();
+                var isLastFrame;
+
+                switch(exportUntil) {
+                    case "tagged":
+                    isLastFrame = (!Object.keys(self.videotagging.frames).length) || (frameId >= parseInt(Object.keys(self.videotagging.frames)[Object.keys(self.videotagging.frames).length-1]));
+                    break;
+                    case "visited":
+                    var lastVisitedFrameId = Math.max.apply(Math, Array.from(self.visitedFrames));
+                    isLastFrame = (frameId >= lastVisitedFrameId);        
+                    break;
+                    case "last":
+                    isLastFrame = (self.videotagging.video.currentTime >= self.videotagging.video.duration);
+                    break;
+                }
+
+                if (isLastFrame) {
+                    self.videotagging.video.oncanplay = null;
+                    resolve();
+                }
+                
+                frameHandler(frameId, frameCanvas, canvasContext, () => {
+                    if (!isLastFrame) {
+                        self.videotagging.stepFwdClicked(false);
+                    }
+                })
+            }
+        });
+    }
+
+    // TODO: Abstract to a module that receives a "framesReader" object as input
+    //exports frames to the selected detection algorithm format  for model training
+    this.export = function(method, exportUntil, exportPath, testSplit, cb) {
+
+        self.detectionAlgorithmManager.setExporter(method, exportPath, self.videotagging.inputtagsarray, 
```

Comment:
Why do you need the fourth argument (e.g. the number of positive frames?)..
It just the number of tagged frames BTW..